cmd/faucet/faucet.go

```diff
@@ -286,7 +306,26 @@ func main() {
 	}
 	pass := strings.TrimSuffix(string(blob), "\n")
 
-	ks := keystore.NewKeyStore(filepath.Join(faucetDirFromConfig(genesis.Config), "keys"), keystore.StandardScryptN, keystore.StandardScryptP)
+	// Get the chain id from the genesis or the designated API.
+	// We'll use this to infer the keystore and chain data directories.
+	// NOTE: Relying on having chain id immediately available may be fragile.
+	// IRC eth_chainId can possibly not return the configured value until the block activating the chain id EIP155 is reached.
+	// See the difference in implementation between ethapi/api.go and eth/api.go #ChainID() methods.
+	// There's an issue open about this somewhere at ethereum/xxx.
+	// This could be resolved by creating a -chainid flag to use as a fallback.
+	chainID := uint64(0)
```

Comment:
:eyes:

Reply:
`-chainid` sounds good to have as a fallback.

Maybe we can request the genesis hash from the attached node and compare it locally? (till EIP155)

Reply:
Yeap, that'll work. I just wrote the code for it, but am now getting the feeling of fix-it-with-a-flag/feature-creep. The only thing this is used for is configuring the data directory namespace. #268 includes a change adding a `-datadir` flag that would provide another way around this potential ambiguity for a user. Otherwise, this ambiguity would only be an issue for users attempting to attach the `faucet` to an (at this point) very-unsynced client. I'm starting to feel inclined to just `if chainID == 0 && { log.Crit... }`.

Reply:
I like the way https://github.com/etclabscore/core-geth/commit/b78074862257c7943e18df3912893a01d7ba4a6e + 1 handles it. Bravo!!!